napari/_qt/qt_viewer.py

```diff
@@ -1028,6 +1029,41 @@ def _set_drag_status(self):
             'Hold <Alt> key to open plugin selection. Hold <Shift> to open files as stack.'
         )
 
+    def _image_from_clipboard(self):
+        """Insert image from clipboard as a new layer if clipboard contains an image or link."""
+        cb = QGuiApplication.clipboard()
+        if cb.mimeData().hasImage():
+            image = cb.image()
+            if image.isNull():
+                return
+            arr = QImg2array(image)
+            self.viewer.add_image(arr)
+            return
+        if cb.mimeData().hasUrls():
+            show_info("No image in clipboard, trying to open link instead.")
+            self._open_from_list_of_urls_data(
+                cb.mimeData().urls(), stack=False, choose_plugin=False
+            )
+            return
+        if cb.mimeData().hasText():
+            show_info(
+                "No image in clipboard, trying to parse text in clipboard as a link."
+            )
+            url_list = []
+            for line in cb.mimeData().text().split("\n"):
+                url = QUrl(line.strip())
```

Comment:
```suggestion
                url = QUrl(line.strip())
                if url.isEmpty():
                	continue
```

I noticed if I copied a URL on a line by selecting the whole line (which would include the end-line character), that I got a `ValueError: PosixPath('.') has an empty name` because the split would occur on `\n` and two `url` would be generated, the one with the URL and an empty one. This check skips in that case.
It also means that a list with an empty line works too!